internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -397,28 +395,40 @@ func (r *PreprovisioningImageReconciler) AddIronicAgentToInfraEnv(ctx context.Co
 		log.Infof("Setting default ironic agent image (%s) for infraEnv %s", ironicAgentImage, infraEnv.Name)
 	}
 
-	conf, err := ignition.GenerateIronicConfig(r.IronicServiceURL, r.IronicInspectorURL, *infraEnvInternal, ironicAgentImage)
+	ironicServiceURL, inspectorURL, err := r.BMOUtils.GetIronicServiceURLS()
 	if err != nil {
-		log.WithError(err).Error("failed to generate Ironic ignition config")
-		return ctrl.Result{}, err
+		log.WithError(err).Error("failed to get IronicServiceURLs")
+		return false, err
 	}
 
-	_, err = r.Installer.UpdateInfraEnvInternal(ctx, installer.UpdateInfraEnvParams{InfraEnvID: *infraEnvInternal.ID, InfraEnvUpdateParams: &models.InfraEnvUpdateParams{}}, swag.String(string(conf)))
+	conf, err := ignition.GenerateIronicConfig(ironicServiceURL, inspectorURL, *infraEnvInternal, ironicAgentImage)
 	if err != nil {
-		return ctrl.Result{}, err
+		log.WithError(err).Error("failed to generate Ironic ignition config")
```

Comment:
Set error status here as well

Reply:
Actually, return an error you can use for status message and set the error in line 141

Reply:
Done